### PR TITLE
Use JSX

### PR DESCRIPTION
**Pull request by wmertens**
Convert the code to use JSX instead of createElement

The code made my eyes water so I did a quick conversion ;)

### CODE REVIEW

**Comment by FezVrasta** [comment]
I don't think JSX is necessary for a so simple piece of code, I prefer to have a cleaner build pipeline and avoid to use custom syntax just for this little gain (that may even not be a gain if we think about the clarity of what's actually happening under the hood).

**Comment by wmertens** [comment]
Well, to me it was not clear that the object was not wrapping the children,
but was next to it. Also, you are already using the Babel React preset so
it just works…

**Comment by FezVrasta** [comment]
Let's keep this PR here and see if other people agree with it.

**Comment by wmertens** [comment]
BTW, it's also marginally more efficient, due to the adding of the props
only on complex components, and using the pre-bound ref functions.

File: src/index.js [hunk 1]
```diff
@@ -57,35 +57,47 @@ export default class ResizeAware extends Component {
...
-        ref: el => (this.resizeElement = el),
-        onLoad: this.handleObjectLoad,
-        'aria-hidden': true,
-        tabIndex: -1,
-      }),
-      Children.map(children, child =>
-        cloneElement(child, !onlyEvent ? {width, height} : null)
-      )
+    if (typeof Component === 'string') {
+      props.ref = this.handleRef;
```
Comment (FezVrasta) [inline src/index.js:77]: Is this going to edit the referenced `props` object properties?

Comment (wmertens) [reply src/index.js:77]: no, props is a new object created by `...props`

Comment (FezVrasta) [reply src/index.js:77]: Yes but the content of it is just a reference of the original one I think? (for any mutable property)

**Comment by wmertens** [comment]
No, it's a new object with entries referring to the originals. By adding
new entries the original does not change.

**Pull request closed without merging**